editor/plugins/script_text_editor.cpp

```diff
@@ -2370,6 +2446,72 @@ void ScriptTextEditor::drop_data_fw(const Point2 &p_point, const Variant &p_data
 	te->grab_focus();
 }
 
+LocalVector<ScriptInstance *> ScriptTextEditor::_get_scripts_for_export_assignment() {
+	LocalVector<ScriptInstance *> script_instances;
+	Node *scene_root = get_tree()->get_edited_scene_root();
+	bool assign_export_variables = scene_root && ClassDB::is_parent_class(script->get_instance_base_type(), "Node");
+
+	if (assign_export_variables) {
+		EditorInspector *inspector = EditorInterface::get_singleton()->get_inspector();
+		if (inspector) {
+			Object *edited_object = inspector->get_edited_object();
+			Node *node_edit = Object::cast_to<Node>(edited_object);
+			MultiNodeEdit *multi_node_edit = Object::cast_to<MultiNodeEdit>(edited_object);
+
+			if (node_edit != nullptr) {
+				ScriptInstance *si = node_edit->get_script_instance();
+				if (si) {
+					script_instances.push_back(si);
+				}
+			} else if (multi_node_edit != nullptr) {
+				Node *es = EditorNode::get_singleton()->get_edited_scene();
+				for (int i = 0; i < multi_node_edit->get_node_count(); i++) {
+					NodePath np = multi_node_edit->get_node(i);
+					Node *node = es->get_node(np);
+					ScriptInstance *si = node->get_script_instance();
+					if (si) {
+						script_instances.push_back(si);
+					}
+				}
+			}
+		}
+
+		// In case there is no current editor selection, it often still makes sense to try to assign the export variable,
+		// So we default to the first node with the script we find in the scene.
+		if (script_instances.is_empty()) {
+			Node *sn = _find_script_node(scene_root, script);
+			if (sn) {
+				script_instances.push_back(sn->get_script_instance());
+			}
+		}
+	}
+
+	return script_instances;
+}
+
+void ScriptTextEditor::_assign_dragged_export_variables() {
+	ERR_FAIL_COND(pending_dragged_exports.is_empty());
+
+	Node *scene_root = get_tree()->get_edited_scene_root();
+
+	// Sanity check, should never be null
+	ERR_FAIL_COND(!scene_root);
+
+	bool export_variable_set = false;
+	for (const DraggedExport &dragged_export : pending_dragged_exports) {
+		bool success = dragged_export.script_instance->set(dragged_export.variable_name, dragged_export.value);
```

Comment:
```suggestion
		if (!dragged_export.script_instance) {
			return;
		}
		bool success = dragged_export.script_instance->set(dragged_export.variable_name, dragged_export.value);
```

This needs to make sure the script instance is still valid, since the user can close the scene before it is assigned and cause a crash.
Actually, I'm not sure if just checking that it is nullptr is enough, since it isn't actually ever set to nullptr. You may need to store the ObjectID instead and then get the object from it and get the script instance on that.